lib/scanner/walk.go

```diff
@@ -62,13 +60,23 @@ type Config struct {
 	EventLogger events.Logger
 	// If ScanOwnership is true, we pick up ownership information on files while scanning.
 	ScanOwnership bool
+	// If ScanXattrs is true, we pick up extended attributes on files while scanning.
+	ScanXattrs bool
+	// Filter for extended attributes
+	XattrFilter XattrFilter
 }
 
 type CurrentFiler interface {
 	// CurrentFile returns the file as seen at last scan.
 	CurrentFile(name string) (protocol.FileInfo, bool)
 }
 
+type XattrFilter interface {
+	Permit(string) bool
+	GetMaxSingleEntrySize() int
+	GetMaxTotalSize() int
```

Comment:
I'd prefer to drop the `Get`

Reply:
So would I, but the attributes are called MaxSingleEntrySize etc. so when interface:ing we need to differentiate